javaslang/src/main/java/javaslang/collection/Vector.java

```diff
@@ -690,7 +684,9 @@ public T get(int index) {
         if ((index < 0) || (index >= length())) {
             throw new IndexOutOfBoundsException("get(" + index + ")");
         } else {
-            return trie.get(index + indexShift).get();
+            final T[] leaf = trie.getLeaf(index);
+            final int leafIndex = lastDigit(trie.offset() + index);
+            return leaf[leafIndex];
```

Comment:
Please put the whole get() method into BMT and replace this with

```java
@Override
public T get(int index) {
    return trie.get(index);
}
```

Additionally the optimized Iterator should also be moved to BMT, and not be part of Vector.

> In addition you can't primitivize a getter (it would have to return an Object), but you can cast an array and get the value out without boxing.

That's right. But we start to mix up our code here in order to do low level optimizations. The primitive optimizations are not part of this PR. We can talk about that later...